src/util/kaldi-holder.cc

```diff
@@ -0,0 +1,74 @@
+// util/kaldi-holder.cc
+
+// Copyright 2009-2011     Microsoft Corporation
+//                2016     Xiaohui Zhang
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "util/kaldi-holder.h"
+#include "matrix/kaldi-matrix.h"
+
+namespace kaldi {
+
+template<class Real>
+bool ExtractObjectRange(const Matrix<Real> &input, const std::string &range,
+                        Matrix<Real> *output) {
+  if (range.empty()) {
+    KALDI_ERR << "Empty range specifier."; 
+    return false;
+  }
+  std::vector<std::string> splits;
+  SplitStringToVector(range, ",", true, &splits);  
+  if (splits.size() > 2) {
+    KALDI_ERR << "Invalid range specifier: " << range;
+    return false;
+  }
+  std::vector<int32> row_range, col_range;
+  SplitStringToIntegers(splits[0], ":", true, &row_range);
```

Comment:
... but you should handle splits[0] == ":" and splits[1] == ":" as special cases.  [passing 'true' is a messy way to do it as it would treat : and :: the same]